// models/playlist/playlist_class.js
var DB = require('models/mysqlConnection');

var PlayList = { 

	getUserPlaylists : function(session, callback)
	{
		var query = 'SELECT * FROM playlists WHERE user_id = "'+session.user_id+'"';
		DB.query(query, function(err, rows, fields)
		{
	  		if (err) 
	  			throw err;
	  		else
	  		{
	  			callback(rows);
	  		}
		});
	},

	getMusicsFromPlaylist : function(playlist_id, session, callback)
	{
		if (playlist_id == "favorites")
			var query = 'SELECT * FROM musics WHERE musics.user_id = "'+session.user_id+'" AND favorite = 1';
		else
			var query = 'SELECT * FROM musics WHERE musics.playlist_id = "'+playlist_id+'"';
		DB.query(query, function(err, rows, fields)
		{
	  		if (err) 
	  			throw err;
	  		else
	  		{
	  			callback(rows);
	  		}
		});
	},

	/**
		params = {id : id, author : author, name : name}
	**/

	editSong : function(params, session, callback)
	{
		var query = 'UPDATE musics SET author = "'+params.author+'", name = "'+params.name+'" WHERE id = "'+params.id+'"';
		DB.query(query, function(err, rows, fields)
		{
	  		if (err) 
	  			throw err;
	  		else
	  		{
	  			callback("ok");
	  		}
		});
	},

	submitNewSong : function(session, song, callback) {

		var type = 1;
		var query = 'INSERT INTO musics (name, url, type_source_id, playlist_id, user_id, author) VALUES ("'+song.name+'", "'+song.url+'", "'+type+'", "'+song.playlist_id+'", "'+session.user_id+'", "'+song.band+'")';
		DB.query(query, function(err, res)
		{
			if (err)
				throw err;
			else
				callback("ok");
		});
	},

	removeSong : function(session, song, callback) {

		var query = 'DELETE FROM musics WHERE musics.id = "'+song.id+'"';
		DB.query(query, function(err, res)
		{
			if (err)
				callback("erreur");
			else
				callback("ok");
		});
	},

	addSongToFavorite : function(session, id_song, callback) {
		var query = 'UPDATE musics SET favorite = 1 WHERE id = "'+id_song+'"';
		DB.query(query, function(err, res)
		{
			if (err)
				callback("erreur");
			else
				callback("ok");
		});
	},

	removeFromFavorite : function(session, id_song, callback) {
		var query = 'UPDATE musics SET favorite = 0 WHERE id = "'+id_song+'"';
		DB.query(query, function(err, res)
		{
			if (err)
				callback("erreur");
			else
				callback("ok");
		});
	}
}

module.exports = PlayList;
